packages/dcos-integration-test/extra/test_legacy_user_management.py

```diff
@@ -188,8 +189,55 @@ def test_dynamic_ui_config(dcos_api_session):
     assert 'uiConfiguration' in data
 
 
-def test_dcos_add_user_script_exists_oss():
+def test_dcos_add_user(dcos_api_session):
     """
-    dcos_add_user.py script exists in /opt/mesosphere/bin directory.
+    dcos_add_user.py script adds a user to IAM using the
+    script dcos_add_user.py.
     """
-    assert os.path.isfile("/opt/mesosphere/bin/dcos_add_user.py")
+
+    email_address = uuid.uuid4().hex + '@example.com'
+    cli = dcos_cli.DcosCli('')
+    command = ['python', '/opt/mesosphere/bin/dcos_add_user.py', email_address]
+    cli.exec_command(command)
+
+    r = dcos_api_session.get('/acs/api/v1/users')
+    r.raise_for_status()
+    expected_user_data = {
+        "uid": email_address,
+        "description": "",
+        "url": "/acs/api/v1/users/" + email_address,
+        "is_remote": True,
+        "is_service": False,
+        "provider_type": "oidc",
+        "provider_id": "https://dcos.auth0.com/"
+    }
+
+    delete_user(dcos_api_session, email_address)
+
+    assert expected_user_data in r.json()['array']
```

Comment:
I know that this is technically correct, but it is odd to assert for user existence after you've deleted the user. I'd swap both lines here and in the following test.

Reply:
I would rather keep it as is. This is so that the test does not leave behind a user if the assert fails. We could do `try...finally` but I thought that this was the neatest option.

Reply:
The problem is that the user doesn't get deleted if anything else fails (e.g. the `raise_for_status`). This is more obvious in the following test, where some `assert`s exist before the `delete_user` and some after.

Could we use a fixture to do the cleanup?

Reply:
Yes. We have https://github.com/mesosphere/dcos-enterprise/blob/master/packages/dcos-integration-test/extra/conftest.py#L164 in DC/OS Enterprise. I considered adding similar to OSS but chose not to for a few reasons:

* It seemed like a bit too much for this one issue - I considered this balance pragmatic, but I am happy to do that work if you like
* The Enterprise helper already has a TODO, stating that it isn't done as someone would like
* There is currently no `dcos_api_session.iam` module, so ideally that would come first - I already plan to make a JIRA for that.

How do you feel if I create a JIRA to add this as follow-up?

Reply:
I certainly don't want you to put much effort in right now. How about going with the `try...finally` option for now?

Reply:
Will do.